packages/server/src/validator/types.ts

```diff
@@ -30,6 +30,18 @@ export const TYPES = {
     test: (param: boolean) => param === true || param === false,
     error: 'Expected boolean type'
   },
+  'blockParams': {
+    test: (param: any) => {
+      if(Object.prototype.toString.call(param) === "[object Object]") {
+        if (param.hasOwnProperty('blockHash')) {
+          return new Validator.BlockHashObject(param).validate();
+        }
+        return new Validator.BlockNumberObject(param).validate();
+      }
+      return /^0[xX]([1-9A-Fa-f]+[0-9A-Fa-f]{0,13}|0)$/.test(param) && Number.MAX_SAFE_INTEGER >= Number(param) || ["earliest", "latest", "pending"].includes(param)
```

Comment:
The regex seems odd to me.  It allows for 12 bytes and the first nibble cannot be 0?  I tested 0x1204567890abcdef123456 for example the regex does not match for me.

Reply:
Maybe also add some of the test cases noted in the EIP for additional coverage

Reply:
First nibble cannot be 0, because block numbers are not starting with 0, that would mean to be padded, which we have to trim afterwards. For example infura doesn't accepts padded blocknumbers, but alchemy accepts them. We can change the regex to accepts, but we may have to trim it afterwards. I mean it could work both ways. 
It allows for this many bytes only, because otherwise it will exceeds javascript max integer size, this is the reason your example is not working. If you cut it down to `0x1204567890abcde`, it should work with this regular expression.

Reply:
I may have "over-insured" in this case. Generally it should be fine to have a longer than 16 character hex for block number, my reason was that at one point where hex values will always convert to JS's MAX_SAFE_INTEGER, but then again it would take quite the long amout of time until we reach that point, so in the end that size limit may be uneeded.

Reply:
Yes, you're right, I'll change the regex to `/^0[xX]([1-9A-Fa-f][0-9A-Fa-f]{0,13}|0)$/`, which will work for our needs.

Reply:
Thanks for double checking.

Reply:
Thank you for the feedback, Luke.